symphony-bdk-core/src/main/java/com/symphony/bdk/core/HttpClient.java

```diff
@@ -0,0 +1,150 @@
+package com.symphony.bdk.core;
+
+import com.symphony.bdk.http.api.ApiClient;
+import com.symphony.bdk.http.api.ApiClientBuilder;
+import com.symphony.bdk.http.api.ApiClientBuilderProvider;
+import com.symphony.bdk.http.api.ApiException;
+import com.symphony.bdk.http.api.ApiResponse;
+import com.symphony.bdk.http.api.Pair;
+import com.symphony.bdk.core.util.ProviderLoader;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.GenericType;
```

Comment:
That should go away at some point right? If we want to be independant?

Reply:
That would be super nice indeed if we could get rif of this classs, maybe by using our own generic type?

Reply:
But the problem is that the generic `ApiClient#invokeAPI` used this `javax.ws.rs.core.GenericType`